torch/csrc/autograd/profiler.cpp

```diff
@@ -410,6 +433,12 @@ thread_event_lists disableProfiler() {
   return state_ptr->consolidate();
 }
 
+void addEventList(std::vector<Event>&& profiledEvents) {
+  auto state_ptr = getProfilerTLSState();
+  TORCH_CHECK(state_ptr, "Profiler must be enabled.");
+  state_ptr->setOrAddRemoteProfiledEvents(std::move(profiledEvents));
```

Comment:
@ilia-cher Refactored to move the events into a field in `ProfilerTLSState` and then use that in consolidation as we discussed.